### ISSUE TITLE
comments

### ISSUE THREAD

**Issue by skeydan** [nstrayer/activity_detection_post#3]
Awesome post - thanks Nick! 
I think it really helps the reader interested in performing a similar task, from data preprocessing to choosing, training and evaluating a suitable model!
I especially like the model checks (accuracy vs. confidence), and I also think that an extremely interesting continuation would be handling streaming data.
In my view it isn't too long either, and the visualizations are not only pretty but add a lot of information to the text!

I'll split up my comments into 3 parts - content, formatting and typos etc.

#### Content

1.  I think I'd stress somewhere - either in the introduction or later, when reporting accuracy - that these are _really_ the raw data, i.e., not even noise filtered (read that on the website). I think that makes the results even more impressive.

2. When restricting the set of activities used, you motivate this in a purely technical way. I find it interesting that we end up with activities that all are in fact activity changes (or transitions): x_to_y. So an alternative (or supplementary, rather, because the technical issue is of course really there) could be to say we're detecting transitions between activities... Just a thought :-)

3. I was wondering what kind of accuracy you get using a bit less dropout... From the learning curves, performance on the validation set is better than on the training set very soon - so the training set is being underfitted, practically... I think I'd prefer reporting a learning curve where the performances are as close as possible, but if I have to  decide I'd like performance on the training set to be just a bit better, so no one says I'm underfitting the training set...

4. For the abstract, I'd perhaps rephrase/reorder it a bit to make it more distant from the title
"In this post, we're training a convolutional neural network to classify changes in physical activity, as measured by accelerometer and gyroscope data from smartphones" or something like that?

#### Formatting

1. I'd perhaps reformat some code blocks so you don't have to use the scroll bar to read the code (especially with the longer blocks)

2. I have no idea but @jjallaire surely knows :-) Wonder if with the new technology, we can provide copy buttons to make copying out code easier?

3. It looks like linked fixed-width text gets very large - like in  " two layer_conv_1ds we "

4. I know I'm being totally pedantic so I'll say  just because I noticed ;-)): there are two different kinds of quotation marks in the model definition ;-) 

#### Typos etc

1. "turn our observation classes from integers to a one-hot, or dummy encoded, vectors" -> one a too much

2. "using a confusion confusion matrix"

**Comment by jjallaire** [comment]
FYI I'm working on a PR now with various minor suggestions (I'll do the code width reduction that Sigrid mentions as part of that).

As regards copying the code, there isn't anything yet in Radix to provide a copy button (good idea though!)

**Comment by nstrayer** [comment]
@skeydan 

These are fantastic comments. I wish I could have had both of you as advisors/editors in previous writing scenarios. As to your points. 

__Content__
1. This is a good point. I didn't even realize they hadn't done any noise filtering. I added a paragraph about this in the data intro section. Having worked with accelerometer data gathering before, i'm actually amazed the model did as well as it did with this being the case. 
2. Added a brief `aside` about this, but it's actually a super interesting thing. Would be interesting to compare an RNN on streaming data to this model scanning for change points and then just predicting the last class it changed to between. 
3. I had always been under the impression that when using dropout, since the training is done on the smaller models you would expect to always get a worse train loss/accuracy than test, just due to it using a non-ensembled model. However, I dropped the rates down and low-and-behold the curves got nice and close to each other and the accuracy spiked consistently into the 90's for percent accuracy. I clearly need to get a bit more familiar with dropout! 
4. I think @jjallaire did a good job of rephrasing it in his PR. 
> Using Keras to train a convolutional neural network to classify physical activity. The dataset was built from the recordings of 30 subjects performing basic activities and postural transitions while carrying a waist-mounted smartphone with embedded inertial sensors.

__Formatting__
1. Again, kindly taken care of by J.J.'s PR (and a good point, i code on way too big of a screen so I never am thoughtful about this.)
2. I'm actually really curious about this and will do a bit of peaking around with the clipboard api to see if it wouldn't be too hard to add to Radix. 
3. Another J.J. fix. 
4. Not pedantic at all! I find that on different keyboards I write different quotes by default and it is super frustrating when the author doesn't have a consistant style! Fixed to be double quotes all the time. 

__Typos__
1. J.J. again doing my job for me. 
2. Yeah, it's a double confusing matrix. Is it a heatmap or is it a matrix? Who knows. (Fixed).

**Comment by nstrayer** [comment]
Changes are pushed in commit https://github.com/nstrayer/activity_detection_post/commit/67a08e4812d4458cf200914192ff4568f306d8dc , which should update the hosted url shortly.

**Comment by nstrayer** [comment]
I forgot to commit the `index_files` so none of the plots were updating. Should be good now.

**Comment by nstrayer** [comment]
Last update. Fixed a run-on sentence in the why-raw-data-is-important/impressive section and also lowered dropout a little more in the convolutional layer and we're up to 94.4% accuracy.

**Comment by jjallaire** [comment]
Re: the clipboard API this is definitely do-able, I do have some questions about how it would visually adorn the code block (since we don't frame it other than the bar on the left margin). I think this is a nice-to-have but I wouldn't sweat it too much as I think many authors won't want the visual business near their code.

**Comment by jjallaire** [comment]
Post is looking terrific! We will import this as soon as we've converted the TF for R blog (will be nice to be able to tweet this post out right away to draw attention to the new blog design/framework).

**Comment by nstrayer** [comment]
A fun and not important aside. I used `callback_lambda` to make an animated confusion matrix showing the training progress. Kinda cool to see how it starts by guessing everything as one class then slowly starts picking out properly. 
![training_progress_250](https://user-images.githubusercontent.com/6764693/42478730-6719e702-839b-11e8-8810-a35ab9a1db2e.gif)

**Comment by skeydan** [comment]
Wow this animation is totally amazing!! Won't say worth an extra post but if you think of the kind of graphics you get in arxiv papers (on the topic of how/what NNs learn) this is soooo much more instructive!
Perhaps one day we do a post together utilizing this (or things like this)?

Ant thanks for addressing my comments! I totally like the added paragraphs. And the learning curve looks perfect now!

Few typos still :-)

- Classsifying in the title ;-)
- utlizing

... and of course, the post overall with the new layout just looks wonderful!

**Comment by jjallaire** [comment]
That is very, very cool! 

One last thing we could consider for the post: if appropriate, a short bibliography and some references. This is honestly mostly about showing off the citation features of Radix so if you don't think a bibliography makes sense or would be too forced then feel free to ignore this idea :-)

**Comment by nstrayer** [comment]
@skeydan fixed the typos (and found a few more)! 

@jjallaire I could definitely put together a brief bibliography.

**Comment by jjallaire** [comment]
Okay, let's do a bibliography (+ appropriate references to it inline) if it isn't too much trouble

**Comment by nstrayer** [comment]
Done. Let me know if you think it's too sparse.

**Comment by jjallaire** [comment]
Looks great! I think we are good to go :-)

Will let you know when the post is live....

**Comment by jjallaire** [comment]
Okay, post is now live here: https://blogs.rstudio.com/tensorflow/posts/2018-07-17-activity-detection/

And Tweeted here: https://twitter.com/rstudio/status/1019215917220319232
